### PR TITLE
(maint) Use new tmpfile for each agent in direct puppet test

### PR DESCRIPTION
**Pull request by whopper**
Previously, the static catalog drift remediation test created
the agent test file path using `agent.tmpdir`, which is really
an alias to `agent[0].tmpdir`. This would normally be fine,
except in the case where the agents array contains both
Linux and Windows hosts, as the temporary directories between
the two differ.

Instead, generate a new tmpfile for each agent in the agent array
and update the test file resource on the master accordingly.

### CODE REVIEW

File: acceptance/tests/direct_puppet/cached_catalog_remediate_local_drift.rb [hunk 1]
```diff
@@ -62,6 +60,19 @@
 
   with_puppet_running_on master, master_opts, basedir do
     agents.each do |agent|
+      agent_test_file_path = agent.tmpfile('foo_file')
+
+      step "Add test file resource to site.pp on master with agent-specific file path" do
+        apply_manifest_on(master, <<MANIFEST, :catch_failures => true)
+        file { "site.pp":
```
Comment (MikaelSmith) [inline acceptance/tests/direct_puppet/cached_catalog_remediate_local_drift.rb]: Good thing this won't be cached.

**Comment by MikaelSmith** [comment]
:+1:

**Comment by MikaelSmith** [comment]
Tried running
```
be rake ci:test:aio SHA=4b37c39a7bd8be31882449415e98172b94d06337 TEST=tests/direct_puppet/cached_catalog_remediate_local_drift.rb TEST_TARGET=centos7-64a-windows2012r2-64a
```

Part of the output
```
  iqv2id07q9z5nnv.delivery.puppetlabs.net (windows2012r2-64-1) 15:36:54$ cat C:/cygwin64/tmp/foo_file.98TLJq
    code_version_2
  iqv2id07q9z5nnv.delivery.puppetlabs.net (windows2012r2-64-1) executed in 0.09 seconds
  Warning:
  Expected: "code_version_1"
    Actual: "code_version_2"
```
Centos ran first, then Windows.

File: acceptance/tests/direct_puppet/cached_catalog_remediate_local_drift.rb [hunk 2]
```diff
@@ -62,6 +60,19 @@
 
   with_puppet_running_on master, master_opts, basedir do
     agents.each do |agent|
+      agent_test_file_path = agent.tmpfile('foo_file')
```
Comment (MikaelSmith) [inline acceptance/tests/direct_puppet/cached_catalog_remediate_local_drift.rb]: We also need to reset foo_file to code_version_1.

Comment (whopper) [inline acceptance/tests/direct_puppet/cached_catalog_remediate_local_drift.rb]: @MikaelSmith updated

**Comment by MikaelSmith** [comment]
This still fails for me. I'm a bit confused.

**Pull request merged**